src/cartesian/Bar.tsx

```diff
@@ -201,18 +201,13 @@ function BarBackground(props: BarBackgroundProps) {
           return null;
         }
 
-        const onMouseEnter = (e: React.MouseEvent<SVGPathElement, MouseEvent>) => {
```

Comment:
These were left this way on purpose so refactoring doesn't need done when we need to add to them

It's easier to have them explicitly defined than to implicitly inherit parameters

Reply:
rather than this
```ts
(event: React.MouseEvent<SVGElement>) => {
    onMouseClickFromProps?.(data, index, event);
    onMouseClickFromContext?.(data, index, event);
    dispatch(
      setActiveClickItemIndex({
        activeIndex: String(index),
        activeDataKey: dataKey,
        activeClickCoordinate: data.tooltipPosition,
      }),
    );
  }
```

```ts
<T extends React.MouseEvent<SVGElement>> (event:T) => {
    onMouseClickFromProps?.(data, index, event);
    onMouseClickFromContext?.(data, index, event);
    dispatch(
      setActiveClickItemIndex({
        activeIndex: String(index),
        activeDataKey: dataKey,
        activeClickCoordinate: data.tooltipPosition,
      }),
    );
  }
```
so now the event can be SVGPathElement MouseEvent

Reply:
I think that might work? We definitely want to be more specific rather than less (prefer the actual element type)

Reply:
```ts 
<T extends SVGElement> (event:React.MouseEvent<T>) => {
    onMouseClickFromProps?.(data, index, event);
    onMouseClickFromContext?.(data, index, event);
    dispatch(
      setActiveClickItemIndex({
        activeIndex: String(index),
        activeDataKey: dataKey,
        activeClickCoordinate: data.tooltipPosition,
      }),
    );
  }
  ```
  
  `onMouseEnterFromContext<SVGPathElement>(entry,i)`  you can use inference or be very explicit